src/snowflake/cli/plugins/snowpark/package_utils.py

```diff
@@ -316,11 +229,34 @@ def check_if_package_is_avaiable_in_conda(package: Requirement, packages: dict)
     return True
 
 
-pip_failed_msg = """pip failed with return code {}.
-            If pip is installed correctly, this may mean you`re trying to install a package
-            that isn't compatible with the host architecture -
-            and generally means it has native libraries."""
+def _perform_native_libraries_check(
+    deps: List[RequirementWithFilesAndDeps], package_native_libraries: PypiOption
+):
+    if _check_for_native_libraries(deps):
+        return (
+            click.confirm(
+                "\n\nWARNING! Some packages appear to have native libraries!\n"
+                "Continue with package installation?",
+                default=False,
+            )
+            if package_native_libraries == PypiOption.ASK
+            else True
+        )
+
+    else:
+        log.info("No non-supported native libraries found in packages (Good news!)...")
+        return True
+
 
-second_chance_msg = """Good news! The following package dependencies can be
-                imported directly from Anaconda, and will be excluded from
-                the zip: {}"""
+def create_zip_name(name: str) -> str:
+    if name.startswith("git+"):
+        return (
+            name.replace("git+https://github.com/", "")
+            .replace(".git", "")
+            .split("/")[-1]
+            + ".zip"
+        )
```

Comment:
Should we consider using regexp? `(?:git\+https://github.com/)(?:\w+/)*(\w+)` Here's also nice tool for testing regexps https://regex101.com/

Reply:
If there's no match we should consider this as "normal", non-git package.